eval.ml

```diff
@@ -8,7 +8,8 @@ type trace = value list
 type environment = binding Env.t
 type heap = heap_value Heap.t
 
-type status = Running | Stopped
+type status = Running | Stopped of value option
```

Comment:
I like stop. It give some additional features. How about `stop expression`. That makes stop and return uniform.